MaxText/layers/llama2.py

```diff
@@ -21,34 +21,116 @@
 import jax.numpy as jnp
 from jax.ad_checkpoint import checkpoint_name
 from jax.sharding import Mesh
-# from jax.experimental.pallas.ops.tpu import flash_attention
 
 from flax import linen as nn
+from flax import nnx
 
 from MaxText.inference import page_manager
 from MaxText.common_types import Config
-from MaxText.layers.linears import mlp_block
+from MaxText.layers.linears import MlpBlock
+from MaxText.layers import initializers
+from MaxText.layers import nnx_wrappers
 from MaxText.layers import quantizations
-from MaxText.layers.attentions import attention_as_linen
+from MaxText.layers.attentions import Attention
 from MaxText.layers.quantizations import AqtQuantization as Quant
-from MaxText.layers.normalizations import rms_norm
-from MaxText.common_types import MODEL_MODE_PREFILL
+from MaxText.layers.normalizations import RMSNorm
+from MaxText.common_types import MODEL_MODE_PREFILL, MODEL_MODE_AUTOREGRESSIVE
 
 
 # -----------------------------------------
 # The Decoder Layer specific for Llama2
 # -----------------------------------------
 
 
-class LlamaDecoderLayer(nn.Module):
+class LlamaDecoderLayer(nnx.Module):
   """Transformer decoder layer that attends to the encoder."""
 
-  config: Config
-  mesh: Mesh
-  model_mode: str
-  quant: Optional[Quant] = None
+  def __init__(
+      self,
+      config: Config,
+      model_mode: str,
+      mesh: Mesh,
+      quant: Optional[Quant] = None,
+      rngs: nnx.Rngs = None,
+  ):
+
+    self.config = config
+    self.mesh = mesh
+    self.quant = quant
+
+    batch_size = 1 if model_mode == MODEL_MODE_PREFILL else config.global_batch_size_to_train_on
+
+    if model_mode == MODEL_MODE_PREFILL:
+      seq_len = config.max_prefill_predict_length
+    elif model_mode == MODEL_MODE_AUTOREGRESSIVE:
+      seq_len = 1
+    else:
+      seq_len = config.max_target_length
+
+    dummy_inputs_shape = (batch_size, seq_len, config.emb_dim)
```

Comment:
These can be different: https://github.com/AI-Hypercomputer/maxtext/blob/f0ffb2d6b57a1a5177a12a6dc253edec20b01d8d/MaxText/pyconfig.py#L590

@gobbleturk do you know the right one to use here? Seems they are the same for me for Llama2-7B

Reply:
Thank you @gobbleturk